src/common/hooks/useAuth.ts

```diff
@@ -0,0 +1,21 @@
+import {useEffect, useState} from 'react';
+import auth, {FirebaseAuthTypes} from '@react-native-firebase/auth';
+
+export function useAuth() {
+  const [user, setUser] = useState<FirebaseAuthTypes.User | null>();
```

Comment:
here it's not just User or null, it's also undefined. So better write
  const [user, setUser] = useState<FirebaseAuthTypes.User | null>(null);